modules/cross_platform.py

```diff
@@ -1,11 +1,42 @@
 def fix_path(path, platform):
-    if platform == "windows":
-        path.replace("/", "\\")
+    if platform == 'windows':
+        path.replace('/', '\\')
     else:
-        path.replace("\\", "/")
+        path.replace('\\', '/')
     return path
 
 def chdir(directory, platform):
     import os
     if os.path.basename(os.getcwd()) != directory:
         os.chdir(fix_path(directory, platform))
+
+def tabprint(array):
+    result = ''
+    length = 0
+    pos = 0
+    # Get length of longest string + 5
+    for item in array:
+        if len(item) > length:
+            length = len(item) + 5
+    # Add items to string to print
+    for item in array:
+        spacer = ''
+        for i in range(length - len(item)):
+            spacer += ' '
+        pos += len(item) + len(spacer)
+        if pos <= term_width() - length:
+            result += item + spacer
+        else:
+            pos = 0
+            result += '\n'
+    print(result)
```

Comment:
I don't understand what kind of items you are appending

Reply:
I am printing an array of names and this takes the strings and makes a table that can be printed.

Reply:
you can find the call here:
https://github.com/YetAnotherPackageInstaller/rewrite/blob/dcdd8dc818a0ff069399239a864a0aeb891cd032/yapi.py#L42
and here:
https://github.com/YetAnotherPackageInstaller/rewrite/blob/dcdd8dc818a0ff069399239a864a0aeb891cd032/yapi.py#L45
the parameter matches is a list of string so in this method the array is intended to be a list of string.
the first for is used to get the max length and add 5 (this will explain below).
the second for create a string and prevent an ugly indentation of the element in the array list.

To be more precise, from this list of string:
```
element one found at
element two not found at path
```
if you have a terminal with a width of 10 the result string will be:
```
element one
  found at
element two
  not found
  at path
```
not print more than 10 characters per line of terminal.